src/bun.js/api/bun/udp_socket.zig

```diff
@@ -0,0 +1,714 @@
+
+        return .undefined;
+    }
+
+    pub fn reload(this: *This, globalThis: *JSGlobalObject, callframe: *CallFrame) callconv(.C) JSValue {
+        const args = callframe.arguments(1);
+
+        if (args.len < 1) {
+            globalThis.throwInvalidArguments("Expected 1 argument", .{});
+            return .zero;
+        }
+
+        const options = args.ptr[0];
+        const config = UDPSocketConfig.fromJS(globalThis, options) orelse {
+            return .zero;
+        };
+
+        config.protect();
+        var previous_config = this.config;
+        previous_config.unprotect();
+        this.config = config;
+
+        return .undefined;
+    }
+
+    pub fn getClosed(this: *This, _: *JSGlobalObject) callconv(.C) JSValue {
+        return JSValue.jsBoolean(this.closed);
+    }
+
+    pub fn getHostname(this: *This, _: *JSGlobalObject) callconv(.C) JSValue {
+        const hostname = JSC.ZigString.init(this.config.hostname);
+        return hostname.toValueGC(this.globalThis);
+    }
+
+    pub fn getPort(this: *This, _: *JSGlobalObject) callconv(.C) JSValue {
+        return JSValue.jsNumber(this.socket.boundPort());
+    }
+
+    fn addressToString(globalThis: *JSGlobalObject, address_bytes: []const u8) JSValue {
+        var text_buf: [512]u8 = undefined;
+        const address: std.net.Address = switch (address_bytes.len) {
+            4 => std.net.Address.initIp4(address_bytes[0..4].*, 0),
+            16 => std.net.Address.initIp6(address_bytes[0..16].*, 0, 0, 0),
+            else => return .undefined,
+        };
+
+        const slice = bun.fmt.formatIp(address, &text_buf) catch unreachable;
+        return bun.String.createLatin1(slice).toJS(globalThis);
+    }
+
+    pub fn getAddress(this: *This, globalThis: *JSGlobalObject) callconv(.C) JSValue {
+        var buf: [64]u8 = [_]u8{0} ** 64;
+        var length: i32 = 64;
+        this.socket.boundIp(&buf, &length);
+
+        const address_bytes = buf[0..@as(usize, @intCast(length))];
+        const port = this.socket.boundPort();
+        return JSSocketAddress__create(
+            globalThis,
+            addressToString(globalThis, address_bytes),
+            @intCast(port),
+            length == 16,
+        );
+    }
+
+    pub fn getRemoteAddress(this: *This, globalThis: *JSC.JSGlobalObject) callconv(.C) JSC.JSValue {
+        const connect_info = this.connect_info orelse return .undefined;
+        var buf: [64]u8 = [_]u8{0} ** 64;
+        var length: i32 = 64;
+        this.socket.remoteIp(&buf, &length);
+
+        const address_bytes = buf[0..@as(usize, @intCast(length))];
+        return JSSocketAddress__create(
+            globalThis,
+            addressToString(globalThis, address_bytes),
+            connect_info.port,
+            length == 16,
+        );
+    }
+
+    pub fn getBinaryType(
+        this: *This,
+        globalThis: *JSGlobalObject,
+    ) callconv(.C) JSValue {
+        return switch (this.config.binary_type) {
```

Comment:
maybe we should add a `label` method to `BinaryType` since i think we've implemented this before